frame/bounties/src/lib.rs

```diff
@@ -613,15 +725,30 @@ decl_module! {
 						// So the council should first unassign the curator, slashing their
 						// deposit.
 						return Err(Error::<T>::PendingPayout.into())
-					}
+					},
 				}
 
-				let bounty_account = Self::bounty_account_id(bounty_id);
+				// Close active subbounty
+				// Subbounty in all states get closed,
+				// except subbounty in "pendingpayout" state.
+				if bounty.active_subbounty.len() > 0 {
```

Comment:
I don't like loop at all. Just allow parent curator to close subbounty and require no active subbounty here.

Reply:
Subbounty closer by Root origin or master curator is already taken care.

```
			// Either `RejectOrigin` or the master curator can close subbounty.
			ensure!(
				maybe_sender.map_or(true, |sender| sender == master_curator),
				BadOrigin
			);
```

Sure it make sense to avoid the iterating over loop here, 

May be can we ensure that all subbounty are closed before closing the parent bounty. like below one.

```
ensure!( bounty.active_subbounty.len() == 0, Error::<T>::SubBountyActive);
```

Closing parent bounty without closing subbounty, may leads to dead state on subbounty. since many subbounty api has dependency on active state of parent bounty.